### DISCUSSION TITLE
Struggling to understand how/when cholorScheme is applied to components

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [mantinedev/mantine#1328] [Q&A] [answered]
Hi!

Just found out about Mantine and I'm loving it, but how the colorScheme is automatically applied to components down the tree is still kind of fuzzy to me.

Docs say:

> All mantine components support dark color scheme natively without any additional steps. To use dark color scheme, wrap your application in MantineProvider and specify colorScheme

I though any component inside that provider would automatically apply the correct text color to contrast with the current background. But it seems the Text component doesn't, at least not directly under the Provider, as per [this example](https://codesandbox.io/s/reverent-morse-f26zhf).

```js
import { Header, MantineProvider, Paper, Text } from "@mantine/core";
import "./styles.css";

export default function App() {
  return (
    <MantineProvider theme={{ colorScheme: "dark" }}>
      <Header height={70}>
        <Text>this text renders in black color</Text>
        <Paper>this text renders in white color</Paper>
      </Header>
    </MantineProvider>
  );
}
```

![image](https://user-images.githubusercontent.com/6718750/165178253-0d52ddf7-7cd4-4352-815e-ad42319b7d83.png)

Am I missing something? Or should I just wrap every Text component I use inside another one (like Paper)?

**Comment by rtivital** [comment, marked as answer]
Text component inherits color from parent element, set `withGlobalStyles` on MantineProvider or color on body element.

**Reply by mkvlrn-cm42** [reply]
That was it. Theming starting to make a lot more sense to me now. Thank you.
